### PR TITLE
StreamingRetry

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/core/azure-core/azure/core/pipeline/transport/aiohttp.py [hunk 1]
```diff
@@ -158,15 +161,20 @@ def _get_request_data(self, request): #pylint: disable=no-self-use
...
 
+    :param pipeline: The pipeline object
+    :param request: The request object
     :param response: The client response object.
     :type response: aiohttp.ClientResponse
     :param block_size: block size of data sent over connection.
     :type block_size: int
     """
-    def __init__(self, response: aiohttp.ClientResponse, block_size: int) -> None:
+    def __init__(self, pipeline, request, response: aiohttp.ClientResponse, block_size: int) -> None:
```
Comment (bryevdv) [inline sdk/core/azure-core/azure/core/pipeline/transport/aiohttp.py]: types for pipeline and request?

File: sdk/core/azure-core/azure/core/pipeline/transport/aiohttp.py [hunk 2]
```diff
@@ -179,15 +187,29 @@ def __len__(self):
...
             except _ResponseStopIteration:
                 self.response.close()
                 raise StopAsyncIteration()
-            except ServiceResponseError:
+            except (ChunkedEncodingError, ConnectionError):
                 retry_total -= 1
                 if retry_total <= 0:
                     retry_active = False
+                else:
+                    await asyncio.sleep(1000)
```
Comment (bryevdv) [inline sdk/core/azure-core/azure/core/pipeline/transport/aiohttp.py]: Maybe store `ONE_SECOND = 1000` somewhere all these pipelines (or `RETRY_INTERVAL = 1000`

File: sdk/core/azure-core/azure/core/pipeline/transport/aiohttp.py [hunk 3]
```diff
@@ -223,7 +245,7 @@ def body(self) -> bytes:
         """Load in memory the body, so it could be accessible from sync methods."""
         self._body = await self.internal_response.read()
 
-    def stream_download(self) -> AsyncIteratorType[bytes]:
+    def stream_download(self, pipeline) -> AsyncIteratorType[bytes]:
```
Comment (bryevdv) [inline sdk/core/azure-core/azure/core/pipeline/transport/aiohttp.py]: type for pipeline?

File: sdk/core/azure-core/azure/core/pipeline/transport/base_async.py [hunk 4]
```diff
@@ -70,7 +70,7 @@ class AsyncHttpResponse(_HttpResponseBase):
 
     Allows for the asynchronous streaming of data from the response.
     """
-    def stream_download(self) -> AsyncIterator[bytes]:
+    def stream_download(self, pipeline) -> AsyncIterator[bytes]:
```
Comment (bryevdv) [inline sdk/core/azure-core/azure/core/pipeline/transport/base_async.py]: type

File: sdk/core/azure-core/azure/core/pipeline/transport/requests_asyncio.py [hunk 5]
```diff
@@ -121,16 +121,21 @@ class AsyncioRequestsTransport(RequestsTransport, AsyncHttpTransport):  # type:
...
 
+    :param pipeline: The pipeline object
+    :param request: The request object
     :param response: The response object.
     :param int block_size: block size of data sent over connection.
     :param generator iter_content_func: Iterator for response data.
     :param int content_length: size of body in bytes.
     """
-    def __init__(self, response: requests.Response, block_size: int) -> None:
+    def __init__(self, pipeline, request, response: requests.Response, block_size: int) -> None:
```
Comment (bryevdv) [inline sdk/core/azure-core/azure/core/pipeline/transport/requests_asyncio.py]: types

**Pull request merged**